### DISCUSSION TITLE
QDrawer Width

### DISCUSSION THREAD

**Discussion by unsalkorkmaz** [quasarframework/quasar#15148] [Ideas / Proposals] [answered]
Hello,

Do we have a chance to get QDrawer Width as a variable.. like those guys;
![Screenshot 2022-12-19 at 21 11 26](https://user-images.githubusercontent.com/207720/208492092-23e0b495-9d36-4724-acfc-57d1b78c72de.png)

or.. how can I export my own variables properly?

Should I add this to app-scss;
```
:root {
    --my_app_left_drawer: 300px;
}
```
and this to quasar.variables.scss;
```
$my_app_left_drawer: 300px;
```

**Comment by yusufkandemir** [comment, marked as answer]
Hi,

QDrawer's `width` is 300px by default. So, unless you set it to another value, it will not change. If you want to change it, however, for example, if you want 500px, you can do `:width="500"`.
https://quasar.dev/layout/drawer#qdrawer-api
![image](https://user-images.githubusercontent.com/6266078/208635487-e38bb70f-ad37-4b44-bd90-d4f9da48a9dd.png)

As for creating CSS/SCSS variables for your own use, you can use both approaches you've shown. If you only use CSS variables, you can set the CSS variable. If you only use SCSS variables, then you can set the SCSS variable. If you might use both, then set both.

If you don't plan to update the drawer width, you can simply do this in `app.scss`:
```scss
:root {
  --left-drawer-width: 300px;
}
```
then in any component:
```html
<style lang="scss" scoped>
.some-class {
  width: calc(var(--left-drawer-width) / 2);
}
</style>
```

If you are planning to update the drawer width, I'd recommend doing this in the layout file:
```html
<template>
  ...
  <q-drawer :width="drawerWidth" ... />
</template>

<script setup>
// ...

const drawerWidth = ref(300);
watch(drawerWidth, (value) => {
  document.body.style.setProperty('--left-drawer-width', `${value}px`);
});

// ...
  drawerWidth.value = 500;
// ...
</script>

<style lang="scss" scoped>
:global(:root) {
  // Unfortunately, this wouldn't work because v-bind is force scoped
  // https://github.com/vuejs/core/issues/7369#issuecomment-1357258624
  // So, you have to use JS to update the CSS variables, see the script section above
  // --left-drawer-width: v-bind(drawerWidth) px;
}
</style>
```
and you can use it the same way as before
